modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -0,0 +1,616 @@
+
+template<typename, typename, typename>
+struct OAKCallHelper;
+
+template<typename Impl, typename... Ins, typename... Outs>
+struct OAKCallHelper<Impl, std::tuple<Ins...>, std::tuple<Outs...> > {
+    template<int... IIs, int... OIs>
+    static void construct_impl(  GOAKContext &ctx
+                               , std::shared_ptr<dai::Node>& node
+                               , std::vector<std::pair<std::string, dai::Buffer>>& in_queues_params
+                               , cv::detail::Seq<IIs...>
+                               , cv::detail::Seq<OIs...>) {
+        Impl::put(ctx.pipeline(),
+                  ctx.camera_size(),
+                  node,
+                  in_queues_params,
+                  get_in<Ins>::get(ctx, IIs)...,
+                  get_out<Outs>::get(ctx, OIs)...);
+    }
+
+    static void construct(GOAKContext &ctx,
+                          std::shared_ptr<dai::Node>& node,
+                          std::vector<std::pair<std::string, dai::Buffer>>& in_queues_params) {
+        construct_impl(ctx,
+                       node,
+                       in_queues_params,
+                       typename cv::detail::MkSeq<sizeof...(Ins)>::type(),
+                       typename cv::detail::MkSeq<sizeof...(Outs)>::type());
+    }
+};
+
+} // namespace detail
+
+struct GOAKKernel {
+    using F = std::function<void(GOAKContext&,
+                                 std::shared_ptr<dai::Node>&,
+                                 std::vector<std::pair<std::string, dai::Buffer>>&)>;
+    explicit GOAKKernel(const F& f) : m_f(f) {}
+    const F m_f;
+};
+
+struct OAKComponent
+{
+    static const char *name() { return "OAK Component"; }
+    GOAKKernel k;
+};
+
+}} // namespace gimpl // namespace cv
+
+using OAKGraph = ade::TypedGraph
+    < cv::gimpl::OAKComponent
+    // FIXME: extend
+    >;
+
+using ConstOAKGraph = ade::ConstTypedGraph
+    < cv::gimpl::OAKComponent
+    // FIXME: extend
+    >;
+
+void cv::gimpl::GOAKExecutable::LinkToParentHelper(ade::NodeHandle handle,
+                                                   const std::vector<ade::NodeHandle>& nodes)
+{
+    ade::NodeHandle parent;
+    for (const auto& indatah : handle.get()->inNodes()) {
+        // indatah - node's input data
+        // need to find which other node produces that data
+        for (const auto& nhp : nodes) {
+            if (m_gm.metadata(nhp).get<NodeType>().t == NodeType::OP) {
+                for (const auto& outdatah : nhp.get()->outNodes()) {
+                    if (indatah == outdatah) {
+                        parent = nhp;
+                    }
+                }
+            }
+        }
+    }
+    GAPI_Assert(m_oak_nodes.find(parent) != m_oak_nodes.end());
+    GAPI_Assert(m_oak_node_inputs[handle].size() ==
+                m_oak_node_outputs[parent].size());
+    for (size_t i = 0; i < m_oak_node_inputs[handle].size(); ++i) {
+        m_oak_node_outputs[parent][i]->link(*(m_oak_node_inputs[handle][i]));
+    }
+}
```

Comment:
I don't quite get what this functions does and why we would need that.

Reply:
It uses G-API graph to find previous operation in graph for the input op handle, then links their inputs/outputs in dai graph